src/playback-watcher.js

```diff
@@ -512,18 +518,36 @@ export default class PlaybackWatcher {
     return false;
   }
 
-  videoUnderflow_(nextRange, buffered, currentTime) {
-    if (nextRange.length === 0) {
+  videoUnderflow_({videoBuffered, audioBuffered, currentTime}) {
+    if (!videoBuffered) {
+      return;
+    }
+    let gap;
+
+    if (videoBuffered.length && audioBuffered.length) {
+      const lastVideoRange = Ranges.findRange(videoBuffered, currentTime - 3);
```

Comment:
why the `-3`?

Reply:
`gapFromVideoUnderflow` implies that audio will keep playing for ~3 seconds during a video underflow and that is what I have seen as well. I will add a comment.

Reply:
Should we add a fudge to it?

Reply:
I'm thinking that we shouldn't as 3 is already a "fudge" here.